parsers/ClinGenDosageSensitivity/src/loadClinGenDosageSensitivity.py

```diff
@@ -0,0 +1,186 @@
+
+
+##############
+# Class: ClinGen Dosage Sensitivity  data loader
+# Desc: Class that loads/parses the ClinGen Dosage Sensitivity  data.
+##############
+class ClinGenDosageSensitivityLoader(SourceDataLoader):
+    source_id: str = "ClinGenDosageSensitivity"
+    provenance_id: str = (
+        "infores:clingen"  # Need to figure out this, can only be filled from one of the values from https://github.com/biolink/biolink-model/blob/master/infores_catalog.yaml
+    )
+    # increment parsing_version whenever changes are made to the parser that would result in changes to parsing output
+    parsing_version: str = "v1.0"
+
+    def __init__(self, test_mode: bool = False, source_data_dir: str = None):
+        """
+        :param test_mode - sets the run into test mode
+        :param source_data_dir - the specific storage directory to save files in
+        """
+        super().__init__(test_mode=test_mode, source_data_dir=source_data_dir)
+        self.cligen_dosage_sensitivity_url = "http://ftp.clinicalgenome.org/"
+        self.cligen_dosage_sensitivity_gene_file = (
+            "ClinGen_gene_curation_list_GRCh38.tsv"
+        )
+        self.clingen_dosage_sensitivity_region_file = (
+            "ClinGen_region_curation_list_GRCh38.tsv"
+        )
+        self.data_files = [
+            self.cligen_dosage_sensitivity_gene_file,
+            self.clingen_dosage_sensitivity_region_file,
+        ]
+
+    def get_latest_source_version(self) -> str:
+        # if possible go to the source and retrieve a string that is the latest version of the source data
+        latest_version = date.today().strftime("%Y%m%d")
+        return latest_version
+
+    def get_data(self) -> bool:
+        # get_data is responsible for fetching the files in self.data_files and saving them to self.data_path
+        data_puller = GetData()
+        for source in self.data_files:
+            source_data_url = f"{self.cligen_dosage_sensitivity_url}{source}"
+            data_puller.pull_via_http(source_data_url, self.data_path)
+        return True
+
+    def parse_data(self) -> dict:
+        """
+        Parses the data file for graph nodes/edges
+
+        :return: ret_val: load_metadata
+        """
+        # This is a made up example of how one might extract nodes and edges from a tsv file
+        # In this case it's taking the subject ID from column 1 and the object ID from column 3,
+        # prepending them with a curie prefix. The predicate comes from column 2. The value in column 4
+        # is set as a property on the edge.
+        extractor = Extractor(file_writer=self.output_file_writer)
+        dosage_sensitivity_gene_file: str = os.path.join(
+            self.data_path, self.cligen_dosage_sensitivity_gene_file
+        )
+        dosage_sensitivity_region_file: str = os.path.join(
+            self.data_path, self.clingen_dosage_sensitivity_region_file
+        )
+        with open(dosage_sensitivity_gene_file, "rt") as fp:
```

Comment:
It is hard to tell if parsing of the gene and regions is exactly the same or where they differ--- this looks like a lot of the same code twice, but the differences in spacing could be obscuring some things. I think it would be worth refactoring the process out to a separate function that could be called once for the gene data and once for the region data.